From: Hannes Reinecke <hare@suse.de>
Date: Thu, 17 Jul 2014 08:19:32 +0200
Subject: scsi_transport_fc: Cap dev_loss_tmo by fast_io_fail
References: bnc#887608
Patch-Mainline: not yet

When fast_io_fail is set dev_loss_tmo must not be set to a lower
value. Doing so is always a user error, so we should refuse to
set it.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/scsi_transport_fc.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/scsi/scsi_transport_fc.c b/drivers/scsi/scsi_transport_fc.c
index 5e1c6dd..5bc39e4 100644
--- a/drivers/scsi/scsi_transport_fc.c
+++ b/drivers/scsi/scsi_transport_fc.c
@@ -883,6 +883,12 @@ static int fc_rport_set_dev_loss_tmo(struct fc_rport *rport,
 	if (rport->fast_io_fail_tmo == -1 &&
 	    val > SCSI_DEVICE_BLOCK_MAX_TIMEOUT)
 		return -EINVAL;
+	/*
+	 * Do not allow a value lower than fast_io_fail
+	 */
+	if (rport->fast_io_fail_tmo != -1 &&
+	    val < rport->fast_io_fail_tmo)
+		return -EINVAL;
 
 	i->f->set_rport_dev_loss_tmo(rport, val);
 	return 0;
-- 
1.7.12.4

